### ISSUE TITLE
ROM won't import if CRC32 hash collides

### ISSUE THREAD

**Issue by DeNickster** [OpenEmu/OpenEmu#3476]
So I've had OpenEmu for almost a year now, and also frequently update it as well.
But when I try to play rom hacks that I've succesfully patched on my patching program, not all of them will work. When I drag the .gba file to openemu it simply won't show up. Anyone who can help me?

P.S. don't know if it's any useful info but I use mac.

**Comment by clobber** [comment]
Provide an example of ROM and patch that will not import. Typically if a file will not import (such as a .GBA file), that means it is already in your library.

**Comment by DeNickster** [comment]
If you mean the patch file and the original file, I will leave them here
![schermafbeelding 2018-01-02 om 01 17 50](https://user-images.githubusercontent.com/35015429/34472120-d1e72de0-ef5a-11e7-816b-87c6e0d8bc14.png)
![schermafbeelding 2018-01-02 om 01 17 24](https://user-images.githubusercontent.com/35015429/34472121-d92b7886-ef5a-11e7-87b5-771ae7914cad.png)

**Comment by DeNickster** [comment]
And I can't drop the files cuz they're not supported, these are the png's.

**Comment by DeNickster** [comment]
Also I triple checked my library and it really wasn't there.

**Comment by clobber** [comment]
There is a CRC32 collision happening between the ROM hack your are importing and another you already have imported. I wouldn't have noticed this had I not already had another Fire Emblem ROM hack imported myself:

```
$ crc32 Fire Emblem - The Road to Ruin v3.0.3.gba
2144df1c
$ crc32 FE6 PME.gba
2144df1c
```
We don't treat CRC32 as a unique identifier for ROMs, only hash it and store as a secondary, but we do throw an error if a match is found during import. We can fix this by not doing that anymore, since a collision is not important for CRC32; only MD5 which is significantly less likely. Thanks for reporting.

**Comment by clobber** [comment]
Just FYI, this is fixed in OpenEmu 2.0.7.
